lib/fund.js

```diff
@@ -37,161 +32,174 @@ fundCmd.completion = function (opts, cb) {
+      if (seenUrls.has(url)) {
+        item = seenUrls.get(url)
+        item.label += `, ${pkgRef}`
+      } else {
+        parent.nodes = (parent.nodes || []).concat(item)
+        seenUrls.set(url, item)
       }
 
-      setStackedItem(funding, result)
-
-      return result
-    }).reduce((res, result) => {
-      // recurse and exclude nodes that are going to be stacked together
-      const { fundingItem } = result
-      const { dependencies, funding } = fundingItem
-      const items = getFundingItems(dependencies)
-      const stackedResult = retrieveStackedItem(funding)
-      items.forEach(i => result.nodes.push(i))
-
-      if (stackedResult && stackedResult !== result) {
-        stackedResult.label += `, ${result.label}`
-        items.forEach(i => stackedResult.nodes.push(i))
-        return res
+      return item
+    },
+    getChildren: (node, nodeResult) => {
+      if (prev.dependencies &&
+        !Object.hasOwnProperty.call(prev.dependencies, node.name)) {
+        prev = node
+        parent = nodeResult
       }
-
-      res.push(result)
-
-      return res
-    }, [])
-
-  const [ result ] = getFundingItems({
-    [fundingInfo.name]: {
-      dependencies: fundingInfo.dependencies,
-      funding: fundingInfo.funding,
-      version: fundingInfo.version
+      return Object.keys(node.dependencies || {})
+        .map(key => ({
+          name: key,
+          ...node.dependencies[key]
+        }))
     }
   })
 
-  return archy(result, '', { unicode: opts.unicode })
+  const [root] = result.nodes
+  return tree(root.label !== header ? { label: header, nodes: [root] } : root)
 }
 
-function openFundingUrl (packageName, cb) {
-  function getUrlAndOpen (packageMetadata) {
-    const { funding } = packageMetadata
-    const { type, url } = retrieveFunding(funding) || {}
-    const noFundingError =
-      new Error(`No funding method available for: ${packageName}`)
-    noFundingError.code = 'ENOFUND'
-    const typePrefix = type ? `${type} funding` : 'Funding'
-    const msg = `${typePrefix} available at the following URL`
-
-    if (validFundingUrl(funding)) {
-      openUrl(url, msg, cb)
+function openFundingUrl ({ actualTree, spec, fundingSourceNumber }) {
+  const retrievePackageMetadata = () => {
+    const arg = npa(spec, actualTree.path)
+    let packageMetadata
+    if (arg.type === 'directory') {
+      if (actualTree.path === arg.where) {
+        packageMetadata = actualTree.package
+      } else {
+        for (const item of actualTree.inventory.values()) {
+          if (item.path === arg.where) {
+            packageMetadata = item.package
+          }
+        }
+      }
     } else {
-      throw noFundingError
+      const [ item ] = [
+        ...actualTree.inventory.values()
+      ]
+        .filter(i => i && i.package &&
+          i.package.name === arg.name &&
+          semver.valid(i.package.version))
+        .sort((a, b) => semver.rcompare(a.package.version, b.package.version))
```

Comment:
This could be golfed down a bit:

```
const nodes = actualTree.inventory.filter(i =>
  i.package.name === arg.name &&
  semver.valid(i.package.version))
const [item] = [...nodes]
  .sort((a, b) => semver.rcompare(a.package.version, b.package.version))
```

Reply:
Puny but non-zero perf improvement by not creating as big an initial array, since the filter happens in the iterator.

Reply:
Another possibility, probably even a little less work, since most nodes in the tree won't have any given name:

```
const [item] = [...actualTree.inventory.query('name', arg.name)]
  .filter(i => semver.valid(i.package.version))
  .sort((a, b) => semver.rcompare(a.package.version, b.package.version))
```